### PR TITLE
3561 - 3

### PR DESCRIPTION
**Pull request by MariaPaharelava**
## Номер
3561

## Номер задания
3

## Ссылка на видео с демо
https://youtu.be/dk8NkdZnwiI

## Комментарии
Все сделала

### CODE REVIEW

File: 3561/3/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.env
+.idea
```
Comment (Moleved) [inline 3561/3/.gitignore]: JFYI: Better way is using [global gitignore](http://egorsmirnov.me/2015/05/04/global-gitignore-file.html) for IDE files

File: 3561/3/controllers/application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# :nodoc:
```
Comment (Vadser) [inline 3561/3/controllers/application_controller.rb]: Why do you need this comment?

File: 3561/3/controllers/application_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
```
Comment (Vadser) [inline 3561/3/controllers/application_controller.rb]: Why do you need this comment?

File: 3561/3/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+                   place_id: params[:id],
+                   user_id: params[:user_id])
+    Place.find(params[:id]).update(rating: calculate_average_rating)
+    redirect_back
+  end
+
+  private
+
+  def calculate_average_rating
+    Comment.where(place_id: params['id']).average(:star).to_f
```
Comment (Vadser) [inline 3561/3/controllers/comments_controller.rb]: Hmmm, looks strange, please read about associations, `@place = Place.find(params['id'])
    @comments = @place.comments`, you have this code in another place, why you can not use it here

File: 3561/3/views/header.erb [hunk 5]
```diff
@@ -0,0 +1,82 @@
+<div>
+  <nav class="w3-sidebar w3-bar-block w3-card w3-top w3-xlarge w3-animate-left" style="display:none;z-index:2;width:20%;min-width:300px" id="mySidebar">
+    <a href="javascript:void(0)" onclick="document.getElementById('mySidebar').style.display='none'" class="w3-bar-item w3-button w3-padding"><i class="fa fa-times fa-fw w3-margin-right"></i>Close
```
Comment (Vadser) [inline 3561/3/views/header.erb]: Please don't use `onclick`, write handler in js, do this in all cases

File: 3561/3/views/header.erb [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+          <button onclick="document.getElementById('login').style.display='none'"
+                  class="w3-button w3-block w3-teal w3-section w3-padding" type="submit">Sing Up
+          </button>
+        </div>
+      </form>
+    </div>
+  </div>
+</div>
+
+<script>
```
Comment (Vadser) [inline 3561/3/views/header.erb]: Move js in separate file

File: 3561/3/views/places/show.erb [hunk 7]
```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Burgerham</title>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+
+  <link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
+  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Raleway">
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (Vadser) [inline 3561/3/views/places/show.erb]: use layouts to not duplicate your code

File: 3561/3/controllers/comments_controller.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+                   place_id: params[:id],
+                   user_id: params[:user_id])
+    Place.find(params[:id]).update(rating: calculate_average_rating)
+    redirect_back
+  end
+
+  private
+
+  def calculate_average_rating
+    Place.find(params['id']).comments.average(:star).to_f
```
Comment (Vadser) [inline 3561/3/controllers/comments_controller.rb]: Why you can not define `place` one time

File: 3561/3/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
+class UsersController < ApplicationController
+  post '/users' do
+    new_user = User.new(name: params[:usrname], email: params[:email])
+    new_user.password = params[:password]
```
Comment (Vadser) [inline 3561/3/controllers/users_controller.rb]: Why you can't define user password with other user params

File: 3561/3/views/header.erb [hunk 10]
```diff
@@ -0,0 +1,70 @@
+<div>
+  <nav class="w3-sidebar w3-bar-block w3-card w3-top w3-xlarge w3-animate-left" style="display:none;z-index:2;width:20%;min-width:300px" id="mySidebar">
+    <a href="javascript:void(0)" onclick="w3_close('mySidebar')" class="w3-bar-item w3-button w3-padding"><i class="fa fa-times fa-fw w3-margin-right"></i>Close
```
Comment (Vadser) [inline 3561/3/views/header.erb]: why i still see onclick events? Write handlers for that in js file

File: 3561/3/controllers/comments_controller.rb [hunk 11]
```diff
@@ -0,0 +1,12 @@
+require_relative '../controllers/application_controller'
+
+class CommentsController < ApplicationController
+  post '/comments' do
+    Comment.create(text: params[:comment],
+                   star: params[:star],
+                   place_id: params[:id],
+                   user_id: params[:user_id])
+    Place.update_place(Place.find(params[:id]))
```
Comment (Vadser) [inline 3561/3/controllers/comments_controller.rb]: Write method for object of this class, it should call like this, `Place.find(params[:id]).update_rating`

File: 3561/3/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+class UsersController < ApplicationController
+  post '/users' do
+    if params[:password] == params[:password_again]
+      new_user = User.new(name: params[:usrname],
+                          email: params[:email],
+                          password: params[:password])
+      save_user_in_session new_user
+    else
+      { message: 'Password not equals', key: 'not_ok' }.to_json
```
Comment (Vadser) [inline 3561/3/controllers/users_controller.rb]: I say you to name key or status one of the keys?

**Pull request merged**